addons/website_slides_survey/models/slide_slide.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class SlidePartnerRelation(models.Model):
+    _inherit = 'slide.slide.partner'
+
+    user_input_ids = fields.One2many('survey.user_input', 'slide_partner_id', 'Certification attempts')
+    survey_quizz_passed = fields.Boolean('Certification Quizz Passed', compute='_compute_survey_quizz_passed')
+
+    @api.depends('partner_id', 'user_input_ids.quizz_passed')
+    def _compute_survey_quizz_passed(self):
+        for record in self:
+            record.survey_quizz_passed = any(user_input.quizz_passed for user_input in record.user_input_ids)
+
+
+class Slide(models.Model):
+    _inherit = 'slide.slide'
+
+    slide_type = fields.Selection(selection_add=[('certification', 'Certification')])
+    survey_id = fields.Many2one('survey.survey', 'Certification')
+
+    _sql_constraints = [
+        ('check_survey_id', "CHECK(slide_type != 'certification' OR survey_id IS NOT NULL)", "A slide of type 'certification' requires a certification."),
+        ('check_certification_preview', "CHECK(slide_type != 'certification' OR is_preview = False)", "A slide of type certification cannot be previewed."),
+    ]
+
+    def _action_set_viewed(self, target_partner):
+        """ If the slide viewed is a certification, we initialize the first survey.user_input
+        for the current partner. """
+        created_slide_partners = super(Slide, self)._action_set_viewed(target_partner)
```

Comment:
new_slide_partners ? We gain 3 characters :D

Reply:
Yes <- this is also 3 characters btw